### PR TITLE
Update docs for vendored plugins

### PR DESCRIPTION
**Pull request by lox**

### CODE REVIEW

**Comment by harrietgrace** [comment]
Just for my own clarification, is a custom plugin consider to be 'local hooks'? Or am I confusing two ideas here 😅

**Comment by lox** [comment]
Apologies for lack of context, I was at the end of my documenting tether :) 

See https://github.com/buildkite/agent/pull/878 for context, I'll write up a better issue description next week!

**Comment by toolmantim** [comment]
I have a question! Does the vendored plugin path have to be relative and starting with `./` (i.e. within your repository), or can it be any path on the filesystem? Say, if you wanted to have plugins sitting on the agent machine somewhere.

**Comment by toolmantim** [comment]
"Repository plugins" as the heading could work! In the leading paragraph we should probably mention the big difference is that you don't specify a version number / SHA like you do for all the other types of plugins. I'll suggest an edit now…

File: pages/pipelines/plugins.md.erb [hunk 1]
```diff
@@ -73,6 +73,20 @@ Branches, tags and commits are all valid after the `#`.
 
 We recommend always specifying a plugin version tag (e.g. `v1.2.3`) to prevent behavior changing unexpectedly, and to prevent stale checkouts of plugins on your agent machines.
 
+
+## Vendored plugins
```
Comment (toolmantim) [inline pages/pipelines/plugins.md.erb]:
```suggestion
## Repository plugins
```

File: pages/pipelines/plugins.md.erb [hunk 2]
```diff
@@ -73,6 +73,20 @@ Branches, tags and commits are all valid after the `#`.
...
+
+You can store plugins within your repository. This is a great way to develop plugins along with code, or to ensure that a plugin doesn't change without you knowing. 
+
+The following example uses the plugin 'my-custom-plugin' that is stored in the repo's `.buildkite` folder:
```
Comment (toolmantim) [inline pages/pipelines/plugins.md.erb]:
```suggestion
The following example uses the plugin 'my-plugin' that is stored in the repository’s `.buildkite/plugins` folder:
```

File: pages/pipelines/plugins.md.erb [hunk 3]
```diff
@@ -73,6 +73,20 @@ Branches, tags and commits are all valid after the `#`.
...
+
+```yml
+steps:
+  - label: "My custom plugin"
```
Comment (toolmantim) [inline pages/pipelines/plugins.md.erb]:
```suggestion
  - label: "A step that uses a repository plugin"
```

File: pages/pipelines/plugins.md.erb [hunk 4]
```diff
@@ -73,6 +73,20 @@ Branches, tags and commits are all valid after the `#`.
...
+    plugins:
+      - ./.buildkite/plugins/my-custom-plugin: ~
```
Comment (toolmantim) [inline pages/pipelines/plugins.md.erb]:
```suggestion
      - ./.buildkite/plugins/my-plugin: ~
```

File: pages/pipelines/plugins.md.erb [hunk 5]
```diff
@@ -73,6 +73,20 @@ Branches, tags and commits are all valid after the `#`.
 
 We recommend always specifying a plugin version tag (e.g. `v1.2.3`) to prevent behavior changing unexpectedly, and to prevent stale checkouts of plugins on your agent machines.
 
+
+## Vendored plugins
+
+You can store plugins within your repository. This is a great way to develop plugins along with code, or to ensure that a plugin doesn't change without you knowing.
```
Comment (toolmantim) [inline pages/pipelines/plugins.md.erb]:
```suggestion
You can store plugins within your repository. This is a great way to develop plugins along with code, or to ensure that a plugin doesn't change without you knowing. 

When referring to a repository plugin in your `pipeline.yml`, there's no need to specify a version as you do with other types of plugins.
```

**Pull request merged**